lib/main.dart

```diff
@@ -1,31 +1,77 @@
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:snack_chat/service/auth/auth_service.dart';
 import 'package:snack_chat/ui/auth/login.dart';
-import 'package:snack_chat/ui/auth/signup.dart';
+import 'package:firebase_core/firebase_core.dart';
+import 'package:snack_chat/ui/auth/login_view_model.dart';
+import 'package:snack_chat/ui/auth/signup_view_model.dart';
+import 'package:snack_chat/ui/chat/chat.dart';
 
-void main() {
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  await Firebase.initializeApp();
   runApp(ChatApp());
 }
 
 class ChatApp extends StatelessWidget {
   // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
-    final primary = Colors.blue;
     final title = "Chat App";
-    return MaterialApp(
-        title: title,
-        theme: ThemeData(
-            brightness: Brightness.dark,
-            primarySwatch: primary,
-            inputDecorationTheme: InputDecorationTheme(
-              focusedBorder: OutlineInputBorder(
-                borderSide: BorderSide(color: primary, width: 2.0),
-                borderRadius: BorderRadius.circular(8.0),
-              ),
-            )),
-        routes: {
-          "/signup": (_) => new SignUpPage(title: title),
-        },
-        home: LoginPage(title: 'Snack Chat'));
+    final primary = Colors.blue;
+
+    final authService = AuthService(FirebaseAuth.instance);
+    authService.signOut();
+    return MultiProvider(
+      //App level dependencies
+      providers: [
+        Provider<AuthService>(
+            create: (_) => authService
+        ),
+        ChangeNotifierProvider<LoginViewModel>(
+            create: (_) => LoginViewModel(authService: authService)
+        ),
+        ChangeNotifierProvider<SignUpViewModel>(
+            create: (_) => SignUpViewModel(authService: authService)
+        ),
+        StreamProvider(
+          create: (context) => context.read<AuthService>().authStateChanges,
```

Comment:
Couldn't you have used `authService.authStateChanges` here to maintain consistency with the precious providers?

Reply:
Ah, yes! Good catch